tests/ui/impl-trait/recursive-coroutine-indirect.rs

```diff
@@ -2,13 +2,15 @@
 //@ ignore-compare-mode-next-solver (explicit revisions)
 //@[next] compile-flags: -Znext-solver
 
-//@[next] build-fail
+//@[next] known-bug: unknown
+//@[next] failure-status: 101
```

Comment:
why does this ICE now?

Reply:
Because we now fail to normalize but otherwise do not emit an error message. 

This should be fixed by my PR which fixes recursive coroutine detection.

Reply:
Fixed by #136073, which I've stacked underneath these changes. Blocked on that PR for now, which is waiting on bors.